examples/uma-server/vasp2.go

```diff
@@ -289,7 +297,8 @@ func (v *Vasp2) handleUmaPayreq(context *gin.Context) {
 	}
 
 	sendingVaspDomain, err := uma.GetVaspDomainFromUmaAddress(request.PayerData.Identifier)
-	if err != nil {
+	addressValidationError := ValidateUmaAddress(request.PayerData.Identifier)
+	if err != nil || addressValidationError != nil {
 		context.JSON(http.StatusBadRequest, gin.H{
 			"status": "ERROR",
 			"reason": fmt.Sprintf("Invalid sender indentifier. UMA address required in the format $alice@vasp.com: %v", err),
```

Comment:
I think this will print an empty error if `err` is nil but `addressValidationError` is not. Maybe just separate branches (even if duplicative)?

Reply:
Ah yeah good catch. Stupid Go error handling